services/functions/linkdex.ts

```diff
@@ -0,0 +1,81 @@
+import { APIGatewayProxyHandlerV2, APIGatewayProxyStructuredResultV2 } from "aws-lambda"
+import { S3Client, GetObjectCommand, HeadObjectCommand, paginateListObjectsV2 } from "@aws-sdk/client-s3"
+import { CarBlockIterator } from '@ipld/car'
+import { LinkIndexer } from 'linkdex'
+
+export const handler: APIGatewayProxyHandlerV2 = async (event) => {
+  const key = event.queryStringParameters?.key ?? ''
+  const bucket = process.env.BUCKET_NAME ?? ''
+  try {
+    return await getLinkdexReport(key, bucket, new S3Client({}))
+  } catch (err: any) {
+    const details = err?.message ?? err
+    return response({ error: { details }}, { statusCode: 500 })
+  }
+}
+
+export async function getLinkdexReport (key: string, bucket: string, s3: S3Client) {
+  if (!key) {
+    return response({ error: { details: 'key query param is required' }}, { statusCode: 400 })
+  }
+  if (!key.startsWith('raw/') || !key.endsWith('.car') || key.split('/').length !== 4) {
+    return response({ error: { details: 'key is forbidden' }}, { statusCode: 403 })
+  }
+  if (!keyExists(key, bucket, s3)) {
+    return response({ error: { details: 'key is forbidden' }}, { statusCode: 403 })
+  }
+  const carKeys = await listSiblings(key, bucket, s3)
+  const index = new LinkIndexer()
+  for (const carKey of carKeys) {
```

Comment:
Add concurrency to reduce runtime and cost...?

Reply:
yes like dis? https://github.com/web3-storage/linkdex-api/pull/1/commits/32baa8d2871caf55ac3415bcf7153b68468be390